src/bindgen/ir/function.rs

```diff
@@ -183,6 +189,51 @@ impl Function {
                 reserved::escape(name);
             }
         }
+
+        // Save the array length of the pointer arguments which need to use
+        // the C-array notation
+        if let Some(tuples) = self.annotations.list("ptrs-as-arrays") {
+            let mut ptrs_as_arrays: HashMap<String, String> = HashMap::new();
+            let mut unnamed_ptrs: Vec<String> = Vec::new();
+            for str_tuple in tuples {
+                let parts: Vec<&str> = str_tuple[1..str_tuple.len() - 1]
+                    .split(';')
+                    .map(|x| x.trim())
+                    .collect();
+                if parts.len() != 2 {
+                    warn!(
+                        "{:?} does not follow the correct syntax, so the annotation is being ignored",
+                        parts
+                    );
+                    continue;
+                }
+                if parts[0] == "_" {
+                    unnamed_ptrs.push(parts[1].to_string());
+                } else {
+                    ptrs_as_arrays.insert(parts[0].to_string(), parts[1].to_string());
+                }
+            }
+
+            for arg in &mut self.args {
+                let (is_annotation, value) = if arg.name.is_none() && unnamed_ptrs.is_empty() {
```

Comment:
Instead of having a hardcoded boolean that we check below and minting an empty value, let's continue? I think this whole loop should be much simpler, something like:

```
if !matches!(arg.ty,Type::Ptr { .. }) {
    continue;
}

let name = match arg.name {
    Some(ref name) => name,
    None => continue,
};

arg.array_length = ptr_as_arrays.get(name).cloned();
```